corehq/apps/auditcare/tasks.py

```diff
@@ -54,14 +54,21 @@ def copy_events_to_sql(limit=1000):
         try:
             kwargs = _pick(doc, ["user", "domain", "ip_address", "session_key",
                                  "headers", "status_code", "user_agent"])
-            kwargs["event_date"] = force_to_datetime(doc.get("event_date"))
+            kwargs.update({
+                "event_date": force_to_datetime(doc.get("event_date")),
+                "couch_id": doc["_id"],
+            })
             if doc["doc_type"] == "NavigationEventAudit":
+                if NavigationEventAudit.objects.filter(couch_id=doc["_id"]).exists():
+                    continue
                 kwargs.update(_pick(doc, ["params", "headers", "status_code", "view", "view_kwargs"]))
                 kwargs.update({
                     "path": doc.get("request_path"),
                 })
                 NavigationEventAudit(**kwargs).save()
```

Comment:
This will be slow and make it hard to do bulk inserts. I think it is possible to combine both of these queries into a single query. One option is to use a common table expression to do both queries in the same SQL statement, which reduces DB round-trip overhead, but is a bit more complex/harder to optimize so will be a bit slower. Another option is make the index on `couch_id` a [partial unique index](https://www.postgresql.org/docs/current/indexes-partial.html#INDEXES-PARTIAL-EX3) ([using Django `postgres_where`](https://stackoverflow.com/a/46785104/10840)), and then use [`INSERT ... ON CONFLICT DO NOTHING`](https://www.postgresql.org/docs/9.5/sql-insert.html#SQL-ON-CONFLICT).